### ISSUE TITLE
Dynamic JS to asset (Vapor request)

### ISSUE THREAD

**Issue by rogierverbrugge** [laravel/nova-issues#2284]
Hi all,

I've just migrated our project to Vapor which works lovely. I noticed the Nova admin area loads quite slow because all the JS is loaded through PHP (think it's checking policies).

Right now it's loading 80 JS files at a time through PHP. Taking 500ms each so AWS lamba charges for 40 seconds. Our admin area is used by lots of people daily which adds up quickly.

Would it be possible to move all the JS to a CDN as that's its intended purpose. I will try to write a plugin for Nova but thought I would let you know.

Kind regards,

Rogier

**Comment by maherelgamil** [comment]
Or we can merge all assets js/css in one file for production

**Comment by jbrooksuk** [comment]
@maherelgamil yeah, you should merge the files into one and edit the layout to fetch that one file.

**Comment by akki-io** [comment]
We use CloudFront for all our assets, we have a large-ish nova (with multiple third-party nova packages) and we did run into similar performance issues. Also, our nova is used by many admins so performance is key.

I was able to abstract out the code and created a small package to generate static JS/CSS. This is somewhat similar to the comment/command here - https://github.com/laravel/nova-issues/issues/1146#issuecomment-515891323 

Feel free to check it out - https://github.com/akki-io/laravel-nova-assets
